detekt-rules-performance/src/test/kotlin/dev/detekt/rules/performance/CouldBeSequenceSpec.kt

```diff
@@ -79,4 +106,76 @@ class CouldBeSequenceSpec(val env: KotlinEnvironmentContainer) {
         """.trimIndent()
         assertThat(subject.lintWithContext(env, code)).isEmpty()
     }
+
+    @Test
+    fun `#8190 - all functions should be accounted for`() {
+        val code = """
+        val bar = listOf<String?>("bar")
+            .filterIsInstance<String>()
+            .filterNot { true }
+            .filterNotNull()
+            .take(1)
+            .drop(1)
+            .takeWhile { true }
+            .mapNotNull { it }
+        """.trimIndent()
+        assertThat(subject.lintWithContext(env, code)).hasSize(1)
+    }
```

Comment:
```suggestion
    @ParameterizedTest
    @ValueSource(strings = [
        ".filterIsInstance<String>()",
        ".filterNot { true }",
        ".filterNotNull()",
        ".take(1)",
        ".drop(1)",
        ".takeWhile { true }",
        ".mapNotNull { it }",
    ])
    fun `#8190 - all functions should be accounted for`(additionalCall: String) {
        val code = """
        val bar = listOf<String>("bar")
            .filter { it.length > 1}
            .map { "text" }
            $additionalCall
        """.trimIndent()
        assertThat(subject.lintWithContext(env, code)).hasSize(1)
    }
```

This will check each method individually.